src/raft/raft.go

```diff
@@ -1124,12 +1124,12 @@ func (rf *Raft) killed() bool {
 func (rf *Raft) dedicatedApplier(applyMsgs [] ApplyMsg) {
 	for _, v := range applyMsgs {
 		rf.applyCh <- v
```

Comment:
TODO: A rewrite is needed here

Reply:
Potential design:
 -> a dedicated channel to notify whenever rf.commitIndex is updated (rf.commitIndex > rf.lastApplied